parts/eventsets/classic_e.lua

```diff
@@ -38,25 +66,105 @@ return {
     noTele=true,
     keyCancel={5,6},
     mesDisp=function(P)
+        local D=P.modeData
+
         setFont(75)
-        GC.mStr(GetLevelStr(P.modeData.lvl),63,210)
+        GC.mStr(GetLevelStr(D.lvl),63,210)
         mText(TEXTOBJ.speedLV,63,290)
-        PLY.draw.drawProgress(P.stat.row,P.modeData.target)
-        if P.modeData.drought>7 then
-            if P.modeData.drought<=14 then
-                GC.setColor(1,1,1,P.modeData.drought/7-1)
+        PLY.draw.drawProgress(P.stat.row,D.target)
+        if D.drought>7 then
+            if D.drought<=14 then
+                GC.setColor(1,1,1,D.drought/7-1)
             else
-                local gb=P.modeData.drought<=21 and 2-P.modeData.drought/14 or .5
+                local gb=D.drought<=21 and 2-D.drought/14 or .5
                 GC.setColor(1,gb,gb)
             end
             setFont(50)
-            GC.mStr(P.modeData.drought,63,130)
+            GC.mStr(D.drought,63,130)
             mDraw(MODES.drought_l.icon,63,200,nil,.5)
         end
+
+        -- Draw Hz counters
+        local spawnHz=(
+            D.firstMoveTimestamp and
+            D.displacement/(D.lastMoveTimestamp-D.spawnTimestamp)
+            or nil
+        )
+        local moveHz=(
+            D.firstMoveTimestamp and
+            D.displacement/(D.lastMoveTimestamp-D.firstMoveTimestamp)
+            or nil
+        )
+        local mistime=(
+            D.firstMoveTimestamp and
+            1000*(D.firstMoveTimestamp-D.spawnTimestamp)
+            or nil
+        )
+
+        local spawnHzStr="---"
+        if spawnHz then
+            if spawnHz<10 then
+                spawnHzStr=string.format("%.2f",spawnHz)
+            elseif spawnHz<100 then
+                spawnHzStr=string.format("%.1f",spawnHz)
+            elseif spawnHz<1e99 then
+                spawnHzStr=string.format("%.0f",spawnHz)
+            else
+                spawnHzStr="∞"
+            end
+        end
+
+        local moveHzStr="---"
+        if moveHz then
+            if moveHz<10 then
+                moveHzStr=string.format("%.2f",moveHz)
+            elseif moveHz<100 then
+                moveHzStr=string.format("%.1f",moveHz)
+            elseif spawnHz<1e99 then
+                moveHzStr=string.format("%.0f",moveHz)
+            else
+                moveHzStr="∞"
+            end
+        end
+
+        local mistimeStr="---"
+        if mistime then
+            if mistime<10 then
+                mistimeStr=string.format("%.2f",mistime)
+            elseif mistime<100 then
+                mistimeStr=string.format("%.1f",mistime)
+            else
+                mistimeStr=string.format("%.0f",mistime)
+            end
+        end
+
+        GC.setColor(getHzColor(spawnHz))
+        setFont(50)
+        GC.mStr(spawnHzStr,30,40)
+
+        GC.setColor(1,1,1)
+        setFont(24)
+        GC.mStr("Hz",110,65)
+
+        setFont(20)
+        GC.mStr(
+            {
```

Comment:
`{ }` appearing in codes that runs every frame can lead to performance issues

Reply:
for the color table `{1,1,1}`, you can simply replace them with pre-defined table like `COLOR.Z`
for the whole string table, you can pre-define them in the file, then change each value before using it